src/react-apps/applications/runtime/src/features/form/data/sagas/submit/index.ts

```diff
@@ -1,28 +1,58 @@
 import { SagaIterator } from 'redux-saga';
-import { call, takeLatest, select } from 'redux-saga/effects';
-import { IRuntimeState } from '../../../../../types';
-import { post } from '../../../../../utils/networking';
+import { call, select, takeLatest } from 'redux-saga/effects';
+
+import WorkflowActions, * as WorkFlowActions from '../../../workflow/actions';
 import FormDataActions from '../../actions';
 import {
-  ISumbitDataAction,
+  ISubmitDataAction,
 } from '../../actions/submit';
-import * as FromDataActionTypes from '../../actions/types';
+import * as FormDataActionTypes from '../../actions/types';
 import { IFormDataState } from '../../reducer';
 
+import { convertDataBindingToModel } from '../../../../../utils/databindings';
+import { put } from '../../../../../utils/networking';
+import { validateFormData } from '../../../../../utils/validation';
+import { IDataModelState } from '../../../datamodell/reducer';
+import { ILayoutState } from '../../../layout/reducer';
+
 const FormDataSelector: (state: any) => IFormDataState = (state: any) => state.formData;
+const DataModelSelector: (state: any) => IDataModelState = (state: any) => state.formDataModel;
+const LayoutSelector: (state: any) => ILayoutState = (state: any) => state.formLayout;
```

Comment:
I think we have typings for state in the "global" typings-file.

Reply:
Right, fixed